### PR TITLE
fix incorrect c10::SymFloat::sqrt

### PR DESCRIPTION
**Pull request by bdhirsh**
Fixes the silent correctness for SDPA in https://github.com/pytorch/pytorch/issues/141710

Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #141728
* #141725

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @kadeng @chauhang @amjames

### CODE REVIEW

**Review by Skylion007 — APPROVED** [review]

**Review by ezyang — APPROVED** [review]

**Comment by drisspg** [comment]
Bad bug 🥲, nice catch. Only affects math party but still bad. I have checked this against many other naive implementations with the scaling is correct and no significant deviations. Since this is applied uniformly across pre softmax scores I wonder if in typical query/key distribution the safe part of softmax makes this a no op. I need to a piece of paper. Anyways nice fix

**Review by drisspg — APPROVED** [review]
